designer/client/src/components/toolbars/scenarioActions/buttons/PerformSingleExecutionButton.tsx

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { loadProcessState } from "../../../../actions/nk";
+import Icon from "../../../../assets/img/toolbarButtons/perform-single-execution.svg";
+import HttpService from "../../../../http/HttpService";
+import { getProcessName, isPerformSingleExecutionPossible, isPerformSingleExecutionVisible } from "../../../../reducers/selectors/graph";
+import { getCapabilities } from "../../../../reducers/selectors/other";
+import { useWindows, WindowKind } from "../../../../windowManager";
+import { ToggleProcessActionModalData } from "../../../modals/DeployProcessDialog";
+import { ToolbarButton } from "../../../toolbarComponents/toolbarButtons";
+import { ToolbarButtonProps } from "../../types";
+import { ACTION_DIALOG_WIDTH } from "../../../../stylesheets/variables";
+import ProcessStateUtils from "../../../Process/ProcessStateUtils";
+import { RootState } from "../../../../reducers";
+import { getProcessState } from "../../../../reducers/selectors/scenarioState";
+import { PredefinedActionName } from "../../../Process/types";
+
+export default function PerformSingleExecutionButton(props: ToolbarButtonProps) {
+    const { t } = useTranslation();
+    const dispatch = useDispatch();
+    const { disabled, type } = props;
+    const scenarioState = useSelector((state: RootState) => getProcessState(state));
+    const isVisible = useSelector(isPerformSingleExecutionVisible);
+    const isPossible = useSelector(isPerformSingleExecutionPossible);
+    const processName = useSelector(getProcessName);
+    const capabilities = useSelector(getCapabilities);
+    const available = !disabled && isPossible && capabilities.deploy;
+
+    const { open } = useWindows();
+    const action = (p, c) => HttpService.performSingleExecution(p, c).finally(() => dispatch(loadProcessState(processName)));
+    const message = t("panels.actions.perform-single-execution.dialog", "Perform single execution", { name: processName });
+
+    const tooltip = ProcessStateUtils.getActionCustomTooltip(scenarioState, PredefinedActionName.PerformSingleExecution) ?? "run now";
```

Comment:
Please use i18n mechanism

Reply:
I think I've done it now. Backend returns type of tooltip ("NOT_ALLOWED_IN_CURRENT_STATE"  /"NOT_ALLOWED_FOR_DEPLOYED_VERSION") and frontend handles actual message rendering with i18s.

Reply:
It's not exactly what I head in mind. I mean  i18n for "run now" text :). I would not apply i18n for backend messages this way. If we decide to do this, it should be done full on BE side and done more holistic. Let's revert this

Reply:
Done, reverted changes related with backed-generated messages and added transaltion for "run now"